C/plugins/storage/postgres/connection.cpp

```diff
@@ -3312,3 +3312,422 @@ const char *p;
+                                                        else
+                                                        {
+                                                        	columnSet.insert(v["column"].GetString());
+                                                        } 
+                                                }
+					}
+				}
+
+				Value& indexes = tables[i]["indexes"];
+				if (!indexes.IsArray())
+                                {
+                                        raiseError("checkSchemaUpdateRequired", "The property return must be an array");
+                                        return false;
+                                }
+
+				for (auto& v : indexes.GetArray())
+                                {
+                                        if (v.IsObject())
+                                        {
+                                                if (v.HasMember("index"))
+                                                {
+                                                        if (!v["index"].IsArray())
+                                                        {
+                                                                Logger::getLogger()->error("%s %d ", __FUNCTION__, __LINE__, "extracting index values , expecting an array here");
+								return false;
+                                                        }
+                                                        else
+                                                        {
+								std::string s;
+								for (auto& i : v["index"].GetArray())
+								{
+									s.append(i.GetString());
+									s.append(",");
+								}
+								// remove last comma
+								if ( s[s.size()-1] == ',')
+								{
+									s.erase(s.size()-1);
+								}
+
+                                                               	indexSet.insert(s);
+                                                        }
+                                                }
+                                        }
+                                }
+
+				tableColumnMap[name] = columnSet;
+				tableIndexMap[name] = indexSet;
+
+			}
+		}
+
+	}
+
+	return true;
+}
+/**
+ * Create scema of tables
+ *
+ * @param payload   The  payload containing information about schema of 
+ *                  tables to create
+ * @return true if the tables can be crated successfully
+ */
+int Connection::create_schema(std::string payload)
+{
+	Logger::getLogger()->error ( "%s:%d payload is  %s", __FUNCTION__, __LINE__, payload.c_str());
+
+	Document document;
+	std::string schema;
+	int version;
+	const char *logSection="CreatingSchema";
+	unsigned long rowsAffectedLastCommand = 0;
+	std::unordered_map<std::string, unordered_set<std::string> > columnMap;
+        std::unordered_map<std::string, unordered_set<std::string> > indexMap;
+
+	bool schemaUpdateRequired = false;
+
+	try {
+                if (payload.empty())
+                {
+                        return -1;
```

Comment:
Probably should log this as we would not be expecting an empty payload

Reply:
Please give more context in the log message. Seeing a log message in syslog that just says "empty payload" doesn't help anybody.